dask_geopandas/core.py

```diff
@@ -778,6 +778,15 @@ def from_dask_dataframe(df, geometry=None):
         the values will be set as 'geometry' column on the GeoDataFrame.
 
     """
+    # If the geometry column is already a partitioned `_Frame`, we can't refer to
+    # it via a keyword-argument due to https://github.com/dask/dask/issues/8308.
+    # Instead, we assign the geometry column using regular dataframe operations,
+    # then refer to that column by name in `map_partitions`.
+    if isinstance(geometry, dd.core._Frame):
+        name = geometry.name or "geometry"
```

Comment:
This will behave incorrectly for falsey names like `0` or `False`. 

```suggestion
        if geometry.name is None:
            name = geometry.name
        else:
            name = "geometry"
```